zbus/src/match_rule_builder.rs

```diff
@@ -200,12 +200,28 @@ impl<'m> MatchRuleBuilder<'m> {
     }
 
     /// Set 0th argument's namespace.
+    #[deprecated = "use arg0ns instead"]
     pub fn arg0namespace<I>(mut self, namespace: I) -> Result<Self>
     where
         I: TryInto<InterfaceName<'m>>,
         I::Error: Into<Error>,
     {
-        self.0.arg0namespace = Some(namespace.try_into().map_err(Into::into)?);
+        let namespace = namespace.try_into().map_err(Into::into)?;
+        self.0.arg0namespace = Some(namespace.clone());
+        self.0.arg0ns = Some(namespace.into());
+
+        Ok(self)
+    }
+
+    /// Set 0th argument's namespace.
+    ///
+    /// The namespace be a valid bus name or a valid element of a bus name.
```

Comment:
We need a bit more docs here. I know, you're just following my bad example here but don't. :laughing: This is the method that most people will use rather than `MatchRule::arg0ns`.

Reply:
Should I list the complete list of requirements?

Reply:
Just copy&paste from the spec and modify a bit?

Reply:
Perhaps a few examples + a link to the spec would be sufficient?

Reply:
> Perhaps a few examples + a link to the spec would be sufficient?

Sure examples would be even better but we should document the rules anyway. It doesn't have to be long +"Please refer to the D-Bus specification [here](link) for details." at the end would do, yeah.

Reply:
For now I left a link to the spec + a few examples of valid and invalid namespaces, similar to all *Name types.

Reply:
> For now I left a link to the spec + a few examples of valid and invalid namespaces, similar to all *Name types.

Looks good. Examples can be make to work for us as tests too (and hence also we ensure the docs remain in sync with the code). See the [thread](https://github.com/dbus2/zbus/pull/385#discussion_r1239866704).

Reply:
Please resolve this thread once you've read my previous comment.